builtin/logical/consul/path_token.go

```diff
@@ -90,26 +108,43 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 	}
 
 	// Create an ACLToken for Consul 1.4 and above
-	policyLink := []*api.ACLTokenPolicyLink{}
-	for _, policyName := range result.Policies {
-		policyLink = append(policyLink, &api.ACLTokenPolicyLink{
-			Name: policyName,
-		})
+	// If policies were supplied here, then overwrite the policies
+	// that were given when the role was written
+	var policyLink []*api.ACLTokenPolicyLink
+	if len(policies) > 0 {
+		policyLink = getPolicies(policies)
+	} else {
+		policyLink = getPolicies(result.Policies)
+	}
+
+	// If a namespace was supplied here, then overwrite the namespace
+	// that was given when the role was written
+	if len(namespace) == 0 {
```

Comment:
Checking for `namespace == ""` is perhaps cleaner and confirms the type to the reader.

Reply:
Agreed

Reply:
Changed https://github.com/hashicorp/vault/pull/13850/commits/45a4d65bcbbef3d2adb5892ed048f5fbf262ad72